data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,197 @@
+                    stack.append(opr[x](a, b))
+                    if verbose:
+                        # output in tabular format
+                        print(
+                            str(x).rjust(8),
+                            ("push(" + str(a) + str(x) + str(b) + ")").ljust(12),
+                            stack,
+                            sep=" | ",
+                        )
+            else:
+                print(f"{x} is neither a number, nor a valid operator")
+                break
+        # If everything executed correctly,the stack will contain
+        # only one element which is the result
+        result = stack[0] if len(stack) == 1 else None
+        return result
+    return 0
+
+
+def is_yes(val: str) -> bool:
+    """
+    Function that checks whether a user has entered any representation of a Yes (y, Y).
+    Any other input is considered as a No.
+
+    Parameters
+    -----------
+    val : str
+        The value entered by user
 
-    # print table header
-    print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
-    print("-" * (30 + len(post_fix)))
-
-    for x in post_fix:
-        if x.isdigit():  # if x in digit
-            stack.append(x)  # append x to stack
-            # output in tabular format
-            print(x.rjust(8), ("push(" + x + ")").ljust(12), ",".join(stack), sep=" | ")
-        else:
-            b = stack.pop()  # pop stack
-            # output in tabular format
-            print("".rjust(8), ("pop(" + b + ")").ljust(12), ",".join(stack), sep=" | ")
-
-            a = stack.pop()  # pop stack
-            # output in tabular format
-            print("".rjust(8), ("pop(" + a + ")").ljust(12), ",".join(stack), sep=" | ")
-
-            stack.append(
-                str(opr[x](int(a), int(b)))
-            )  # evaluate the 2 values popped from stack & push result to stack
-            # output in tabular format
-            print(
-                x.rjust(8),
-                ("push(" + a + x + b + ")").ljust(12),
-                ",".join(stack),
-                sep=" | ",
-            )
-
-    return int(stack[0])
+    Returns
+    -------
+    bool
+        True if Yes, otherwise False
+    """
+    return val in ["Y", "y"]
 
 
 if __name__ == "__main__":
-    Postfix = input("\n\nEnter a Postfix Equation (space separated) = ").split(" ")
-    print("\n\tResult = ", solve(Postfix))
+    loop = True
+    # Creating a loop so that user can evaluate postfix expression multiple times
+    while loop:
+        expression = input("Enter a Postfix Expression (space separated): ").split(" ")
+        choice = input("Do you want to see stack contents while evaluating? [y/N]: ")
+        display = is_yes(choice)
+        output = evaluate(expression, display)
+        if output is not None:
+            print("Result = ", output)
+        choice = input("Do you want to enter another expression? [y/N]: ")
+        loop = is_yes(choice)
```

Comment:
You can just use an infinite loop with a break to avoid the need for an extra helper variable:
```suggestion
    # Creating a loop so that user can evaluate postfix expression multiple times
    while True:
        expression = input("Enter a Postfix Expression (space separated): ").split(" ")
        choice = input("Do you want to see stack contents while evaluating? [y/N]: ")
        display = is_yes(choice)
        output = evaluate(expression, display)
        print("Result = ", output)
        choice = input("Do you want to enter another expression? [y/N]: ")
        if is_yes(choice):
             break
```